src/audio/pipeline.c

```diff
@@ -215,23 +215,25 @@ int pipeline_free(struct pipeline *p)
 
 	trace_pipe_with_ids(p, "pipeline_free()");
 
+	data.start = p->source_comp;
+
 	/* make sure we are not in use */
-	if (p->source_comp->state > COMP_STATE_READY) {
-		trace_pipe_error_with_ids(p, "pipeline_free() error: Pipeline"
-					  " in use, %u, %u",
-					  p->source_comp->comp.id,
-					  p->source_comp->state);
-		return -EBUSY;
+	if (p->source_comp) {
+		if (p->source_comp->state > COMP_STATE_READY) {
```

Comment:
@bkokoszx can we remove the second if and add it into the first one with a &&?

Reply:
@ranj063 My intention was to return EBUSY only when p->source_comp->state > COMP_STATE_READY. In other case, when p->source_comp is not null we can free components resources.